pyanalyze/name_check_visitor.py

```diff
@@ -3431,17 +3461,35 @@ def _visit_assert_errors_block(self, node: ast.With) -> None:
             )
 
     def visit_AsyncWith(self, node: ast.AsyncWith) -> None:
-        for item in node.items:
-            self.visit_withitem(item, is_async=True)
-        self._generic_visit_list(node.body)
+        self.visit_single_cm(node.items, node.body, True)
+
+    def visit_single_cm(
+        self, items: List[ast.withitem], body: Iterable[ast.AST], is_async: bool = False
```

Comment:
```suggestion
        self, items: List[ast.withitem], body: Iterable[ast.AST], *, is_async: bool = False
```
Cleaner to make this keyword-only